docusaurus/docs/how-to-guides/data-source-plugins/profiling-backend-plugin.md

```diff
@@ -0,0 +1,121 @@
+
+```shell
+INFO [07-09|19:15:00] Profiling enabled   logger=plugin.<plugin ID> blockProfileRate=1 mutexProfileRate=1
+```
+
+Check what debugging endpoints are available by browsing `http://<host>:<profiling_port>/debug/pprof`.
+
+There are some additional [godeltaprof](https://github.com/grafana/pyroscope-go/tree/main/godeltaprof) endpoints available which are more suitable in a continuous profiling scenario. These endpoints are `/debug/pprof/delta_heap`, `/debug/pprof/delta_block`, `/debug/pprof/delta_mutex`.
+
+:::note
+
+To be able to use `profiling_block_rate` and `profiling_mutex_rate` your plugin needs to use at least [`grafana-plugin-sdk-go v0.238.0`](https://github.com/grafana/grafana-plugin-sdk-go/releases/tag/v0.238.0). Refer to [Update the Go SDK](../../create-a-plugin/develop-a-plugin/work-with-backend) for update instructions.
+
+:::
+
+### profiling_enabled
+
+Enable/disable profiling. Default `false`.
+
+### profiling_port
+
+Optionally customize the HTTP port where profile data is exposed, if for example profiling multiple plugins or the default port is taken. Default `6060`.
+
+### profiling_block_rate
+
+:::note
+
+The higher the fraction (that is, the smaller this value) the more overhead it adds to normal operations.
+
+:::
+
+Controls the fraction of goroutine blocking events that are reported in the blocking profile, default `0` (i.e. track no events). Using `5` would report 20% of all events as an example. See https://pkg.go.dev/runtime#SetBlockProfileRate for more detailed information.
+
+:::note
+
+The higher the fraction (that is, the smaller this value) the more overhead it adds to normal operations.
+
+:::
+
+### profiling_mutex_rate
+
+Controls the fraction of mutex contention events that are reported in the mutex profile, default `0` (i.e. track no events). Using `5` would report 20% of all events as an example. See https://pkg.go.dev/runtime#SetMutexProfileFraction for more detailed information.
+
+:::note
+
+The higher the fraction (that is, the smaller this value) the more overhead it adds to normal operations.
+
+:::
+
+## Collect and analyze profiles
+
+In general, you use the [Go command pprof](https://golang.org/cmd/pprof/) to both collect and analyze profiling data. You can also use [curl](https://curl.se/) or similar to collect profiles which could be convenient in environments where you don't have the Go/pprof command available. Next, some usage examples of using curl and pprof to collect and analyze memory and CPU profiles.
+
+### Analyzing high memory usage/memory leaks
+
+When experiencing high memory usage or potential memory leaks it's useful to collect several heap profiles and later when analyzing, compare them. It's a good idea to wait some time, e.g. 30 seconds, between collecting each profile to allow memory consumption to increase.
+
+```bash
+curl http://<profile-addr>:<profile-port>/debug/pprof/heap > heap1.pprof
```

Comment:
you were using `http://<host>:<profile-port>` before, I guess better to use the same in all the examples. Also, it'd be good to clarify what this host is, the Grafana URL? Something else?